product_esc_code/models/product_product.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2022 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    esc_code = fields.Char(related="product_tmpl_id.esc_code")
```

Comment:
```suggestion
```
Not necessary to do this since product.product inherits product.template.

Technically speaking, however, we should follow how `default_code` fields are defined in these models.

https://github.com/odoo/odoo/blob/943e5b9761b43257ec968c12f15723c618c3b5ad/addons/product/models/product_product.py#L31

https://github.com/odoo/odoo/blob/02cdc5218eef422106bf43246286afaa0475b67c/addons/product/models/product_template.py#L127-L129

The client currently does not use variants, therefore it is OK to simply have the field in product.template, unless the situation changes in the future (while this module is in use).